src/app/models/function.rb

```diff
@@ -6,11 +6,30 @@ class Function < ApplicationRecord
 
   belongs_to :user
 
+  before_validation :function_name
+
   def to_param
     name
   end
 
   def self.from_param(param)
     find_by(name: param)
   end
+
+  private
+
+  def walk_ast(ast)
```

Comment:
[R] we need to give this function a meaningful name that reflect what it does, these are the names I'm thinking about:
- extract_function_name
- find_first_function
- find_name

Reply:
I'd like to go forward with find_first_function